lib/ansible/plugins/strategy/linear.py

```diff
@@ -391,7 +391,7 @@ def run(self, iterator, play_context):
                 if any_errors_fatal and (len(failed_hosts) > 0 or len(unreachable_hosts) > 0):
                     for host in hosts_left:
                         (s, _) = iterator.get_next_task_for_host(host, peek=True)
-                        if s.run_state != iterator.ITERATING_RESCUE or \
+                        if s.run_state not in [iterator.ITERATING_RESCUE, iterator.ITERATING_ALWAYS] or \
```

Comment:
Is s.run_state a bitfield like s.fail-state?  If so:
``` python
if s.run_state & (iterator.ITERATING_RESCUE | iterator.ITERATING_ALWAYS) or \
```

You could also do the combining of ITERATING_RESCUE | ITERATING_ALWAYS outside of the loop but I'm not sure if that's a pre-mature optimization or not.